src/components/forms/MemoryLimitAvailable.tsx

```diff
@@ -1,59 +1,134 @@
-
-  const {
-    data: resources,
-    error,
-    isLoading,
-  } = useQuery({
-    queryKey: [queryKeys.resources],
-    queryFn: async () => fetchResources(),
-    enabled: canViewResources(),
-  });
+  const isClustered = useIsClustered();
+  const values = formik?.values;
+  const projectResourceMemory = limitToBytes(project?.config["limits.memory"]);
+  const { target } = values as CreateInstanceFormValues;
+  const { location } = values as EditInstanceFormValues;
+  const isClusterGroupTarget = target?.startsWith(CLUSTER_GROUP_PREFIX);
+  const validTarget = target && !isClusterGroupTarget ? target : undefined;
+  const clusterMember = isClustered ? (validTarget ?? location) : undefined;
+
+  const { data: resources, error, isLoading } = useResources(clusterMember);
 
   if (isLoading) {
     return <Spinner className="u-loader" text="Loading resources..." />;
   }
-
   if (error) {
     notify.failure("Loading resources failed", error);
+    return null;
   }
+  if (!resources) {
+    return null;
+  }
+
+  const getSourceQualifier = (source: string) => {
+    if (source === "project") {
+      return (
+        <>
+          {"in project "}
+          <ResourceLink
+            type="project"
+            value={project?.name || "default"}
+            to={`/ui/projects/${encodeURIComponent(project?.name || "default")}`}
+          />
+        </>
+      );
+    }
 
-  const getAvailableMemory = () => {
-    if (!project?.config["limits.memory"]) {
-      return resources?.memory.total;
+    if (source === "cluster-member") {
+      return (
+        <>
+          {"on cluster member "}
+          <ResourceLink
+            type="cluster-member"
+            value={clusterMember ?? ""}
+            to={`/ui/cluster/member/${encodeURIComponent(clusterMember ?? "")}`}
+          />
+        </>
+      );
     }
-    if (!resources?.memory.total) {
-      return limitToBytes(project.config["limits.memory"]);
+    return null;
+  };
+
+  const getAvailableMemory = (): MemoryLimit => {
+    const resourceArray = Array.isArray(resources) ? resources : [resources];
+    const projectLimit = Number(projectResourceMemory);
+
+    if (resourceArray.length === 0) {
+      return {
+        min: projectLimit,
+        max: projectLimit,
+        source: projectLimit ? "project" : "",
+      };
     }
-    return Math.min(
-      resources.memory.total,
-      Number(limitToBytes(project.config["limits.memory"])),
-    );
+    const resourcesMemories = resourceArray.map((r) => r.memory.total);
+    const minResourceMemory = Math.min(...resourcesMemories);
+    const maxResourceMemory = Math.max(...resourcesMemories);
+    if (!projectResourceMemory) {
+      return {
+        min: minResourceMemory,
+        max: maxResourceMemory,
+        source: clusterMember ? "cluster-member" : "",
+      };
+    }
+
+    return {
+      min: Math.min(minResourceMemory, projectLimit),
+      max: Math.min(maxResourceMemory, projectLimit),
+      source: "",
```

Comment:
```suggestion
      min: minResourceMemory,
      max: projectLimit,
      source: "",
```
I think all other cases are handled above now. 

The remaining case is a wired edge. The project limit is between two cluster member resources. I think it is OK to not report where the limit comes from in that case, though.

Reply:
Yes, there are a couple of edge cases that would make the code more complicated and it is not worth it to report the limit in that case.

Reply:
Sounds good to me.
I think we should simplify by removing the Math.min / Math.max cases though, to make clear which edge case remains here.